app/geo/geo.component.ts

```diff
@@ -0,0 +1,65 @@
+import {Component, OnInit, ElementRef, ViewChild} from '@angular/core';
```

Comment:
Talking about architecture I would like to note that for now all the main functionality is glued together in geo.component and most of all the components do not have any logic - just template rendering

Solution may be following:
```
<wapi-navbar></wapi-navbar>

<wapi-jumbotron></wapi-jumbotron>

<main class="main">
    <wapi-geo *ngIf="position" [position]="position"></wapi-geo> 

    <hr/>

    <wapi-google-map *ngIf="position" [posisition]="position"></wapi-google-map>
</main>

<wapi-footer></wapi-footer>
```

adding ```@Input() position``` to map component and weather component, and in the main app component making a geoposition request and pass the position to components, where it will be used in ngOnInit to render necessary elements (probably also calling some service methods)

probably after those changes it makes sence to rename ```wapi-geo``` to ```wapi-weather``` or smth

also ngOnChanges hook can be used during component lifetime if position is going to change

as a result you will have independent **reusable** components